core/io/resource.h

```diff
@@ -81,13 +81,14 @@ class Resource : public RefCounted {
 	void _take_over_path(const String &p_path);
 
 public:
-	static Node *(*_get_local_scene_func)(); //used by editor
-	static void (*_update_configuration_warning)(); //used by editor
+	static Node *(*_get_local_scene_func)(); //Used by editor.
+	static void (*_update_configuration_warning)(); //Used by editor.
+	static Ref<Resource> get_remap_resource(const Ref<Resource> &p_resource, HashMap<Ref<Resource>, Ref<Resource>> &remap_cache, const Ref<Resource> &p_fallback, Node *p_for_scene);
 
 	void update_configuration_warning();
 	virtual bool editor_can_reload_from_file();
-	virtual void reset_state(); //for resources that use variable amount of properties, either via _validate_property or _get_property_list, this function needs to be implemented to correctly clear state
-	virtual Error copy_from(const Ref<Resource> &p_resource);
+	virtual void reset_state(); //For resources that use variable amount of properties, either via _validate_property or _get_property_list, this function needs to be implemented to correctly clear state.
+	virtual Error copy_from(const Ref<Resource> &p_resource, HashMap<Ref<Resource>, Ref<Resource>> &remap_cache, bool p_keep_local_to_scene_resource_instance = false);
```

Comment:
I would honestly make a different function like `copy_from_with_remap` or something like this, so other places that don't need to use this logic don't  get the code polluted unnecesarily. In fact, if this is only used in packed_scene, I would just make this a local function of packed_scene.